### DISCUSSION TITLE
What is the order for "type -a"?

### DISCUSSION THREAD

**Discussion by YousufSSyed** [fish-shell/fish-shell#10297] [Q&A] [answered]
This is the output when I do `type -a type`:
```
type is a builtin
type is /usr/bin/type
```
Does fish sort them in any particular order? Which `type` runs when I write it?

**Comment by faho** [comment, marked as answer]
[Functions beat builtins beat commands](https://fishshell.com/docs/current/language.html#command-lookup), so that's the order type uses.

The first one it prints is also the one that runs, unless you decorate it like `command type` (which runs /usr/bin/type) or `builtin type` (which runs the builtin, ignoring any functions if you have any).

If `type -a type` ran the type *command* it could not tell you about the builtins - they only exist inside the shell
